be/src/vec/functions/like.cpp

```diff
@@ -129,6 +130,42 @@ Status FunctionLikeBase::vector_vector(const ColumnString::Chars& values,
                                        const ColumnString::Offsets& pattern_offsets,
                                        ColumnUInt8::Container& result, const LikeFn& function,
                                        LikeSearchState* search_state) {
+    // for constant_substring_fn, use long run length search for performance
+    if (constant_substring_fn ==
+        *(function.target<doris::Status (*)(LikeSearchState * state, const StringValue&,
+                                            const StringValue&, unsigned char*)>())) {
+        // treat continous multi string data as a long string data
+        const UInt8* begin = values.data();
+        const UInt8* end = begin + values.size();
+        const UInt8* pos = begin;
+
+        /// Current index in the array of strings.
+        size_t i = 0;
+        size_t needle_size = search_state->substring_pattern.get_pattern_length();
+
+        /// We will search for the next occurrence in all strings at once.
+        while (pos < end) {
+            // search return matched substring start offset
+            pos = (UInt8*)search_state->substring_pattern.search((char*)pos, end - pos);
+            if (pos >= end) break;
+
+            /// Determine which index it refers to.
+            /// begin + value_offsets[i] is the start offset of string at i+1
+            while (begin + value_offsets[i] <= pos) ++i;
+
+            /// We check that the entry does not pass through the boundaries of strings.
```

Comment:
Whether our `text` and `pattern` may contain `\0`? 
If it is impossible to contain `\0`, then this check will be unnecessary

Reply:
It's necessary to check the boundary.

Here is an example:

pattern: "cdef"
row1: "abcd"
row2: "efgh"

values.data: "abcdefgh"
value_offsets: [4, 8]

values.data "abcdefgh" will match pattern "cdef" at pos 2. The end of matched substring is pos+needle_size == 6. It's beyond the end offset of row1, which is begin+value_offsets[0]. So due to the check 'if (pos + needle_size < begin + value_offsets[i])', we will not set result[i]=1 improperly.

Reply:
`values.data` will be "abcd\0efgh" here.
But after my investigation, there may be `\0` in the input string, this check is necessary.

Reply:
OK. thanks for you careful review!